tokio-quiche/src/quic/io/worker.rs

```diff
@@ -475,26 +495,53 @@ where
         let mut send_buf = &mut send_buf[self.write_state.bytes_written..];
         if send_buf.len() > segment_size.unwrap_or(usize::MAX) {
             // Never let the buffer be longer than segment size, for GSO to
-            // function properly
+            // function properly.
             send_buf = &mut send_buf[..segment_size.unwrap_or(usize::MAX)];
         }
 
-        match qconn.send(send_buf) {
+        // On the first call to `select_path()` a path will be chosen based on
+        // the local address the connection initially landed on. Once a path is
+        // selected following calls to `select_path()` will return it, until it
+        // is reset at the start of the next write cycle.
```

Comment:
> Once a path is selected following calls to `select_path()` will return it, until it
> is reset at the start of the next write cycle.

How important is it that select_path return the same path?

Reply:
I still don't understand the issue TBH... Are we trying to protect against compiler bugs? Otherwise I don't see how the scenario you are describing is possible. And if we do that here, then we should add `debug_assert!()` to every other `if` block? Like, why is this case special.

Reply:
Spoke offline. We have tests that will break if we remove the early return.